hoodie-client/src/main/java/com/uber/hoodie/io/compact/strategy/DayBasedCompactionStrategy.java

```diff
@@ -57,8 +58,22 @@
   @Override
   public List<CompactionOperation> orderAndFilter(HoodieWriteConfig writeConfig,
       List<CompactionOperation> operations) {
-    // Iterate through the operations and accept operations as long as we are within the IO limit
-    return super.orderAndFilter(writeConfig,
-        operations.stream().sorted(comparator).collect(Collectors.toList()));
+    // Iterate through the operations and accept operations as long as we are within the configured target partitions
+    // limit
+    int numPartitions = 0;
+    String lastPartitionPath = "";
+    List<CompactionOperation> filteredList = new ArrayList<>();
+    List<CompactionOperation> sortedOperations = operations.stream().sorted(comparator).collect(Collectors.toList());
```

Comment:
can't we do `stream.limit(N)` directly, instead of tracking num partitions.

Reply:
here, I need to compare the compactionOperations partition paths and then limit, could not find the java 8 way of doing that.

Reply:
Here is one example to limit 10 partitions: 

operations.stream().collect(Collectors.groupingBy(HoodieCompactionOperation::getPartitionPath)).entrySet().stream().sorted(
        Map.Entry.comparingByKey(comparator2)).limit(10).flatMap(e -> e.getValue().stream()).collect(Collectors.toList());

Here comparator2 is similar to existing comparator but operates on partitionPath directly instead of CompactionOperation